Cubical/Categories/Bifunctor/Redundant.agda

```diff
@@ -0,0 +1,560 @@
+{-
+
+  Explicit "redundant" definition of a bifunctor F : C , D → E.
+  Includes all 3 of
+  1. A "parallel action" f × g
+  2. Two "separate actions" f × d and c × g
+  Plus axioms that say
+  1. The actions are functorial
+  2. and that they agree
+
+  The full definition includes some redundant *axioms* to make it
+  convenient to use, so we include two alternative formulations to
```

Comment:
The convenience here relates to better definitional equality.

The quick sell is that we would like to act functorially only on one side of `C ×C D` without additional identities tossed in the resulting term. If you had some `F : Functor (C ×C D) E`, in regular math `F` would reduce to a functorial action on `f : C [ c , c' ]` via  `F .F-hom (f , D .id)`, but this introduces a composition with an unnecessary identity. So we build these redundant axioms to give a native action that isn't over some `D .⋆IdR`/`D .⋆IdL` path. 

I'm eliding a fair bit of detail here, but this definitional behavior is crucial for the comprehension construction.

Like we discussed, this PR will be split out into a few separate ones by topic and one of them will just be these bifunctor/binproduct definitions and some motivating examples.

Reply:
> I'm eliding a fair bit of detail here, but this definitional behavior is crucial for the comprehension construction.

Specifically, it's crucial for making certain compositional constructions spit out the "right" definitions. The main example where it came up for us was when we defined the exponential `a -> -` as a right adjoint to `a x -`, the universal property for exponentials involved an annoying `id` because we were fixing the product functor on one side. We'll be sure to include more info about this in comments when we make the non-draft PR